### PR TITLE
Updated test vectors

### PR DESCRIPTION
**Pull request by vivek-arte**
This PR updates the test vectors to be of version 6 (as opposed to version 7). 

It also provides a new file for the V6 test vectors, an improvement to having them in the same file but behind feature flags.

It provides a error for when the asset description string is not a UTF8 encoding.

### CODE REVIEW

File: zcash_primitives/src/transaction/tests.rs [hunk 1]
```diff
@@ -313,7 +314,14 @@ fn zip_0244() {
         (tdata, txdata.digest(TxIdDigester))
     }
 
-    for tv in self::data::zip_0244::make_test_vectors() {
+    #[cfg(not(zcash_unstable = "nu6" /* TODO nu7 */ ))]
+    let test_vectors = self::data::zip_0244::make_test_vectors();
+    #[cfg(zcash_unstable = "nu6" /* TODO nu7 */ )]
+    let mut test_vectors = self::data::zip_0244::make_test_vectors();
```
Comment (alexeykoren) [inline zcash_primitives/src/transaction/tests.rs:320]: Are we doing all this just to avoid 'mut' in case of not nu6? Seems like too much to me

Comment (vivek-arte) [reply zcash_primitives/src/transaction/tests.rs:320]: Yeah, should I use `#[allow(unused_mut)]` instead?

**Review by PaulLaux — APPROVED** [review]

File: zcash_primitives/src/transaction/components/issuance.rs [hunk 2]
```diff
@@ -50,7 +50,8 @@ fn read_authorization<R: Read>(mut reader: R) -> io::Result<Signed> {
 
 fn read_action<R: Read>(mut reader: R) -> io::Result<IssueAction> {
     let asset_descr_bytes = Vector::read(&mut reader, |r| r.read_u8())?;
-    let asset_descr: String = asset_descr_bytes.iter().map(|&b| b as char).collect();
+    let asset_descr: String = String::from_utf8(asset_descr_bytes)
+        .map_err(|_| Error::new(ErrorKind::InvalidData, "Asset Description not valid UTF-8"))?;
```
[the remark this answers is not in the document]
Comment (vivek-arte) [reply zcash_primitives/src/transaction/components/issuance.rs:55]: Added comment.

File: zcash_primitives/src/transaction/tests.rs (hunk 1 above)
Comment (vivek-arte) [reply zcash_primitives/src/transaction/tests.rs:318]: Done.

**Pull request merged**